tests/test-lexer.cpp

```diff
@@ -147,18 +145,69 @@ TEST_CASE("Test the Schema class", "[Schema]") {
 
     SECTION("Test repetition regex") {
         // Repetition without capture groups untagged and tagged AST are the same
-        test_regex_ast("a{0,10}", U"()|(a{1,10})");
-        test_regex_ast("a{5,10}", U"a{5,10}");
-        test_regex_ast("a*", U"()|(a{1,inf})");
-        test_regex_ast("a+", U"a{1,inf}");
+        test_regex_ast("capture:a{0,10}", U"()|((a){1,10})");
+        test_regex_ast("capture:a{5,10}", U"(a){5,10}");
+        test_regex_ast("capture:a*", U"()|((a){1,inf})");
+        test_regex_ast("capture:a+", U"(a){1,inf}");
 
         // Repetition with capture groups untagged and tagged AST are different
-        test_regex_ast("(?<letter>a){0,10}", U"(<~0>)|((a)<0>{1,10})");
-        test_regex_ast("(?<letter>a){5,10}", U"(a)<0>{5,10}");
-        test_regex_ast("(?<letter>a)*", U"(<~0>)|((a)<0>{1,inf})");
-        test_regex_ast("(?<letter>a)+", U"(a)<0>{1,inf}");
+        test_regex_ast("capture:(?<letter>a){0,10}", U"(<~0>)|(((a)<0>){1,10})");
+        test_regex_ast("capture:(?<letter>a){5,10}", U"((a)<0>){5,10}");
+        test_regex_ast("capture:(?<letter>a)*", U"(<~0>)|(((a)<0>){1,inf})");
+        test_regex_ast("capture:(?<letter>a)+", U"((a)<0>){1,inf}");
 
         // Capture group with repetition
-        test_regex_ast("(?<letter>a{0,10})", U"(()|(a{1,10}))<0>");
+        test_regex_ast("capture:(?<letter>a{0,10})", U"(()|((a){1,10}))<0>");
+
+        // Complex repetition
+        test_regex_ast(
+                // clang-format off
+                "capture:"
+                "("
+                    "("
+                        "(?<letterA>a)|"
+                        "(?<letterB>b)"
+                    ")*"
+                ")|("
+                    "("
+                        "(?<letterC>c)|"
+                        "(?<letterD>d)"
+                    "){0,10}"
+                ")",
+                U"("
+                    U"(<~0><~1>)|(("
+                        U"((a)<0><~1>)|"
+                        U"((b)<1><~0>)"
+                    U"){1,inf})"
+                U"<~2><~3>)|("
+                    U"(<~2><~3>)|(("
+                        U"((c)<2><~3>)|"
+                        U"((d)<3><~2>)"
+                    U"){1,10})"
+                U"<~0><~1>)"
+                // clang-format on
+        );
     }
 }
+
+namespace {
+auto test_regex_ast(string_view const var_schema, u32string const& expected_serialized_ast)
+        -> void {
+    log_surgeon::Schema schema;
+    schema.add_variable(var_schema, -1);
+    auto const schema_ast = schema.release_schema_ast_ptr();
+    auto const* capture_rule_ast = dynamic_cast<SchemaVarAST*>(schema_ast->m_schema_vars[0].get());
+    REQUIRE(capture_rule_ast != nullptr);
+
+    auto const actual_string
+            = fmt::format("{}", u32string_to_string(capture_rule_ast->m_regex_ptr->serialize()));
+    auto const expected_string = fmt::format("{}", u32string_to_string(expected_serialized_ast));
```

Comment:
I feel like `fmt::format` isn't necessary here.
```suggestion
    auto const actual_string = u32string_to_string(capture_rule_ast->m_regex_ptr->serialize());
    auto const expected_string = u32string_to_string(expected_serialized_ast);
```